### ISSUE TITLE
Moderation

### ISSUE THREAD

**Issue by DIWesser** [tunerapp/radiobrowser-server#9]
I think I should bring the various comments on moderation floating around in my head into one spot and not keep sprinkling them through the other issues. I've roughly sorted these by how soon we'll have to deal with them.

- [ ] Deduplicating current data set
Now that I've looked at the data a bit more, this seems like it will be fairly straight forward if time consuming. The almost-dupes tool (#6) probably isn't worth putting effort into unless we find it's needed when we/I've gotten through the obvious duplicates

- [ ] Station error issues
Not everyone who notices a entry is broken is going to know how to fix it. Issues seem like a good way to handle this. But keeping on top of this would require a fair bit off attention. (Worth noting that UUIDs seem to be the easiest way to link radio-browser.info entries with our repo https://github.com/tunerapp/radiobrowser-server/issues/3#issuecomment-806266173)

- [ ] Pull requests for changes
Do we want to automatically accept changes to the data set? If so, we want some sort of bot that compares current entries, checks things like account age and a blacklist, waits an hour or two, then approves pull requests for us. It might be worth looking at whatever validation radio-browser.info is currently doing

- [ ] Moderation policies
At some point someone will probably link to some pirated content which we'll have to remove. Reading between the lines, it sounds like [Alex has had problems](https://github.com/segler-alex/radiobrowser-api-rust/issues/49#issuecomment-629766589) with nazi stations.

- [ ] Broken channels
radio-browser.info automatically detects broken urls. Do we want some way to have an issue automatically generated if a url is broken for more than a day or two? If so, should we have a way to archive entries or just delete them if we find that the station has shut down?

**Comment by louis77** [comment]
>     * [ ]  Deduplicating current data set
>       Now that I've looked at the data a bit more, this seems like it will be fairly straight forward if time consuming. The almost-dupes tool (#6) probably isn't worth putting effort into unless we find it's needed when we/I've gotten through the obvious duplicates

Yes, we don't have to be perfect from the beginning. Running some SQL to clean-up duplicated content might be a good start. I'd like to add that the "tags" are a real mess right now, up to the point where they are completly useless. I think it would be nice if we could clean that up too and maybe even decide on a fixed set of tags which can be used. What do you think?

>     * [ ]  Station error issues
>       Not everyone who notices a entry is broken is going to know how to fix it. Issues seem like a good way to handle this. But keeping on top of this would require a fair bit off attention. (Worth noting that UUIDs seem to be the easiest way to link radio-browser.info entries with our repo [#3 (comment)](https://github.com/tunerapp/radiobrowser-server/issues/3#issuecomment-806266173))

Since you mention _Broken channels_ below, I assume that you talk about station metadata like broken favicons, urls etc?

>     * [ ]  Pull requests for changes
>       Do we want to automatically accept changes to the data set? If so, we want some sort of bot that compares current entries, checks things like account age and a blacklist, waits an hour or two, then approves pull requests for us. It might be worth looking at whatever validation radio-browser.info is currently doing

I don't think that meta data about the GitHub account the change is coming from will do as any good. The philosophy that made radio-browser.info great in the first place is that anyone could submit new stations without special validation (of course we should automatically check if the Favicon, Homepage etc. are returning HTTP 200). There are downsides but for the moment I would say that having a GitHub account might be an entry barrier high enough. 

>     * [ ]  Templates
>       Whatever we decide for issues and pull requests, we'll probably need some templates

Yes, that would be great.

>     * [ ]  Moderation policies
>       At some point someone will probably link to some pirated content which we'll have to remove. Reading between the lines, it sounds like [Alex has had problems](https://github.com/segler-alex/radiobrowser-api-rust/issues/49#issuecomment-629766589) with nazi stations. Any thoughts on anything else we should preemptively not allow?

My thoughts on this are, that we shouldn't be the ones who decide what is allowed and appropriate and what is not. What might be legal and appropriate in the US might be forbidden in Germany and the other way around. What I could think of is that we will introduce a flag like `sensitive: true` API users can decide to exclude these stations altogether. What do you think?

>     * [ ]  Broken channels
>       radio-browser.info automatically detects broken urls. Do we want some way to have an issue automatically generated if a url is broken for more than a day or two? If so, should we have a way to archive entries or just delete them if we find that the station has shut down?

There is the `LastCheckOkTime` from upstream so we might use that for as long as we implement our own checking alogrithm.
